### PR TITLE
Enhancements and bug fixes to the discovery page

### LINKED ISSUE #2023: discover page shows zero stars for all notebooks
this information is not being shared; it's stored in editor_tab.js, which takes care of the notebook tree and the inventory.

### LINKED ISSUE #2039: Thumbnail image disappears from 'Discover' page after any changes in 'Assets' div
Steps :
- Create a new notebook and save after adding contents to it
- Upload thumb.png image to assets div
- Load Discover page (the thumbnail image will be displayed on Discover page for that notebook)
- Now load the same notebook again and make any change in assets div e.g edit Scratch.R, add new asset, delete asset, etc.
- Open Discover page (thumbnail image will disappear)

The thumbnail image will not be displayed even if we undo the changes made in 'Assets' div.

### PR DESCRIPTION
**Pull request by paulinshek**
Added an extra check that thumb.png.b64 is seen before testing for nullity, this should fix #2039, and #2011 

Discover page now shows the number of forks, to address #2030
- However, should clicking on the fork icon create a new fork of that notebook? I don't think clicking on the star adds to favourites?

Also, I have fixed #2023 and the star counts are now showing.

Finally, I also added calls to rcloud.config.set.recently.modified.notebook when a  notebook is forked or created, so now all newly created or forked notebooks show up on discover page recently modified,  to address #2032 and #2028

### CODE REVIEW

**Comment by paulinshek** [comment]
Actually, the fork count is still always zero- will keep working on this!

File: rcloud.support/R/ocaps.R [hunk 1]
```diff
@@ -137,7 +138,7 @@ unauthenticated.ocaps <- function(mode, compute)
       signal_to_compute = make.oc(.signal.to.compute),
       help = compute$help,
       get_thumb = compute$get_thumb,
-
+      get_fork_count = compute$get_fork_count,
```
Comment (gordonwoodhull) [inline rcloud.support/R/ocaps.R]: We should revisit if these need to be run from compute process or if they can be moved to the control process. If it works for now, that's fine, but I think it's generally better to put them on control if possible. (They will wait for cells to execute, and I think we eventually want to move all RCS access to control for security reasons.)

**Comment by gordonwoodhull** [comment]
As for the fork count, this looks correct, pulling the count from RCS. The issue is probably that we need to populate `forkcount` with the value for all notebooks, since it's a new key and the source of the info is GitHub. 

We can create a notebook to do the population; the only tricky thing is running against [GitHub rate limits](https://developer.github.com/v3/#rate-limiting). Reading info about every existing notebook is a big deal.

**Comment by gordonwoodhull** [comment]
In the meantime, you can test by refreshing the fork count on particular notebooks by.. forking them. They have to belong to someone else for it to be a "real fork", though, i.e. you may want a fake account.

File: rcloud.support/R/rcloud.support.R [hunk 2]
```diff
@@ -407,14 +410,17 @@ rcloud.fork.notebook <- function(id, source = NULL) {
                                      list(owner=owner,
                                           description=src.nb$content$description,
                                           id=src.nb$content$id))
-    } else ## src=dst, regular fork
+    } else {## src=dst, regular fork
         new.nb <- fork.gist(id, ctx = src.ctx)
-
+        rcloud.config.set.recently.modified.notebook(id, new.nb$content$updated_at)
```
Comment (gordonwoodhull) [inline rcloud.support/R/rcloud.support.R]: This is creating an entry for the *source* notebook with the updated date of the *fork*. I'll open a new issue for this.

**Comment by gordonwoodhull** [comment]
Great work, thanks @paulinshek!

For future commits, please use the phrases described here
https://help.github.com/articles/closing-issues-via-commit-messages/

in your commit messages, so that the issue will be automatically closed when the code is merged.

Rebased and merged.

**Comment by paulinshek** [comment]
Thanks @gordonwoodhull, will do!

**Pull request closed without merging**